pipelines/google/google_gemini.py

```diff
@@ -561,29 +576,17 @@ async def pipe(
                 return "Error: No valid message content found"
                 
             # Configure generation parameters and safety settings
-            generation_config, safety_settings = self._configure_generation(body)
-            
-            # Initialize the model with system instruction
-            client = genai.GenerativeModel(
-                model_name=model_id,
-                system_instruction=system_instruction
-            )
+            generation_config = self._configure_generation(body, system_instruction)
             
             # Make the API call
+            client = self._get_client()
             if stream:
-                # For streaming response we'll still use the synchronous method but wrap it
-                # in an async call using a thread pool executor (handled by retry_with_backoff)
                 try:
                     async def get_streaming_response():
-                        loop = asyncio.get_event_loop()
-                        return await loop.run_in_executor(
-                            None,
-                            lambda: client.generate_content(
-                                contents,
-                                generation_config=generation_config,
-                                safety_settings=safety_settings,
-                                stream=True,
-                            )
+                        return await client.aio.models.generate_content_stream(
+                            model=model_id,
+                            contents=contents,
+                            config=generation_config,

```

Comment:
Switched to the async method. was there a particular reason to use the sync version?